src/cocotb/task.py

```diff
@@ -210,21 +212,18 @@ def cancel(self, msg: Optional[str] = None) -> None:
         if self.done():
```

Comment:
As long as the state is updated after running the Task with `CancelledError`, calling `cancel()` again will override the already set `_cancelled_error`. The order of state update and running the Task will have to be reviewed in your scheduling followup PR.